app/Models/Season.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+declare(strict_types=1);
+
+namespace App\Models;
+
+use Carbon\Carbon;
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Relations\BelongsToMany;
+
+/**
+ * @property bool $concluded
+ * @property string $name
+ * @property-read Collection<Multiplayer\Room> $rooms
+ */
+class Season extends Model
+{
+    use HasFactory;
+
+    protected $casts = [
+        'concluded' => 'boolean',
+    ];
+
+    public function endDate(): ?Carbon
+    {
+        return $this->concluded
+            ? $this->rooms->max('ends_at')
```

Comment:
basically, spotlights seasons contain of few playlists created one after another so it would end up on on having end date changed with each playlist added. the concluded attribute would be set to true after the season conclude, so after last lobby is added.